NavigationReactNative/src/TabBar.tsx

```diff
@@ -54,9 +55,19 @@ class TabBar extends React.Component<any, any> {
         var TabBar = primary ? NVTabBar : TabBarPager;
         var TabView = primary ? NVTabNavigation : (!I18nManager.isRTL ? NVTabLayout : NVTabLayoutRTL);
         TabView = Platform.OS === 'android' ? TabView : NVSegmentedTab;
+
+        var visibilityMode = labelVisibility;
+
+        if(Platform.OS === 'android'){
+            var constants = (UIManager as any).getViewManagerConfig('NVTabNavigation').Constants;
+            visibilityMode = (labelVisibility === 'selected' && tabBarItems.length > 3) ? constants.labelVisibility['auto'] : constants.labelVisibility[labelVisibility]
+        }
+
         var tabLayout = (Platform.OS === 'android' || !primary) && (
             <TabView
                 bottomTabs={bottomTabs}
+                labelVisibility={visibilityMode}
```

Comment:
I prefer the following. What do you think?
```jsx
labelVisibility = !(labelVisibility === 'selected' && tabBarItems.length > 3) ? labelVisibility : 'auto';
<TabView
  labelVisibility={Platform.OS === 'android' ? constants.labelVisibility[labelVisibility] : null}
```

Reply:
I was trying to find a way to make this a bit more concise but Monday evening brain just wasn't doing it :) Will use your suggestion here. 

I would suggest assigning this to a new variable rather than overriding the input prop though, we need to keep a reference for the `itemHorizontalTranslation` prop. Unless we compare again using constants, but I feel that's overcomplicating things. What do you think?

Reply:
Don't think we need a new variable. The `itemHorizontalTranslation` will be true when we change `selected` to `auto` so should be fine, right?

Reply:
If tabs < 4 we want to keep 'selected', and will need to turn off `itemHorizontalTranslation`. I could do something like 

`constants.labelVisibility[labelVisibility] !== constants.labelVisibility['selected']`

But I feel like just keeping a reference to the original prop makes more sense here.

Reply:
🤦  I've just realised on second look that we are not setting the constant as a variable. I'll fix that now.